contracts/strategies/liquity/LiquityLQTYStakingStrategy.sol

```diff
@@ -0,0 +1,311 @@
+     *      governance can still trigger an emergency exit, in order to
+     *      pause investing in the pool.
+     */
+    function emergencyExit() public onlyGovernance {
+        _setPausedInvesting(true);
+    }
+
+    /**
+     * @dev Resumes the ability to invest into the underlying rewards pool.
+     */
+    function continueInvesting() public onlyGovernance {
+        _setPausedInvesting(false);
+    }
+
+    /**
+     * @dev Can completely disable liquidation of DODO rewards. This can be
+     *      useful in case of emergency exits.
+     */
+    function setSell(bool _sell) public onlyGovernance {
+        _setSell(_sell);
+    }
+
+    /**
+     * @dev Set the minimum amount of DODO needed to trigger a sale (for a liquidation).
+     */
+    function setSellFloor(uint256 _sellFloor) public onlyGovernance {
+        setSellFloor(_sellFloor);
+    }
+
+    function finalizeUpgrade() external onlyGovernance {
+        _finalizeUpgrade();
+    }
+
+    // ---------------- Internal methods ----------------
+
+    function rewardPoolBalance() internal view returns (uint256 balance) {
+        balance = ILQTYStaking(rewardPool()).stakes(address(this));
+    }
+
+    function claimRewards() internal {
+        if (rewardPoolBalance() > 0) {
+            ILQTYStaking(rewardPool()).unstake(0);
+        }
+    }
+
+    function enterRewardPool() internal {
+        uint256 currentBalance = IERC20(underlying()).balanceOf(address(this));
+        ILQTYStaking(rewardPool()).stake(currentBalance);
+    }
+
+    function exitRewardPool() internal {
+        uint256 staked = rewardPoolBalance();
+        if (staked > 0) {
+            ILQTYStaking(rewardPool()).unstake(staked);
+        }
+    }
+
+    /**
+     * @dev Stakes everything the strategy holds into the reward pool.
+     */
+    function investAllUnderlying() internal onlyNotPausedInvesting {
+        // This check is needed for avoiding reverts in case we stake 0
+        if (IERC20(underlying()).balanceOf(address(this)) > 0) {
+            enterRewardPool();
+        }
+    }
+
+    function _liquidateReward() internal {
+        // Profits can be disabled for possible simplified and rapid exit
+        if (!sell()) {
+            emit ProfitsNotCollected(sell(), true);
+            return;
+        }
+
+        uint256 lusdBalance = IERC20(lusd).balanceOf(address(this));
+        uint256 ethBalance = address(this).balance;
+
+        // Liquidate the LUSD rewards to USDC
+        if (lusdBalance > 0) {
+            IERC20(lusd).safeApprove(lusd3CrvPair, 0);
+            IERC20(lusd).safeApprove(lusd3CrvPair, lusdBalance);
+            ICurveMetapool(lusd3CrvPair).exchange_underlying(
+                0,
+                2,
+                lusdBalance,
+                1
+            );
+        }
+
+        // Liquidate the ETH rewards to USDC
+        if (ethBalance > 0) {
+            ISwapRouter(uniswapV3Router).exactInputSingle.value(ethBalance)(
```

Comment:
Could we give the universal liquidator a try? This has a more convenient interface. The id for the dex is "uniV3".

https://github.com/harvest-finance/harvest-strategy/blob/master/contracts/strategies/balancer/BalancerStrategy.sol#L111?

Reply:
Is the code for the UniswapV3 DEX contract public (eg. https://etherscan.io/address/0x1D35ba854575B576B3C0aB4e64E27Bf2D2c1D48A)? Not sure how to specify the path since the UniversalLiquidator contract (eg. https://etherscan.io/address/0x875680a120597732f92bf649cacfeb308e54dba4) accepts an array of addresses as the path while UniswapV3 deals with the path as bytes?

Reply:
Some examples are in our `FeeRewardForwarder` that swaps a reward token into FARM: https://etherscan.io/address/0x153C544f72329c1ba521DDf5086cf2fA98C86676#code
```
    storedLiquidationPaths[sushi][farm] = [sushi, weth, farm];
    storedLiquidationDexes[sushi][farm] = [sushiDex, uniDex];
```
- This basically means that `SUSHI` is liquidated into `WETH` using `Sushiswap` (`sushiDex`), and `WETH` liquidated into `FARM` using `Uniswap` (`uniDex`). Dexes are bytes32 which are defined as:

```
  bytes32 public uniDex = bytes32(uint256(keccak256("uni")));
  bytes32 public sushiDex = bytes32(uint256(keccak256("sushi")));
```

and for uniswap V3, you could use:

```
bytes32 public uniV3Dex = bytes32(uint256(keccak256("uniV3")));
```

Reply:
Thanks, that was very clear!